app/models/user.rb

```diff
@@ -1,3 +1,5 @@
 class User < ActiveRecord::Base
-	has_secure_password
+	  has_secure_password
+    validates :username, :email, :password_digest, :presence => true
+    validates :email, uniqueness: true
```

Comment:
username should also be unique

Reply:
In the sign-up page, when the email is already taken, the input box is shrunk.  I think you may want to keep the box and the content as in the original form, so the user can type in and see the what email he typed in.

Reply:
Yeah, thats still a bug that happens when Rails returns an error. It modifies the input CSS and wrecks everything. I'm still looking for a way to fix it.

Reply:
Fixed.